src/distributor-detector.ts

```diff
@@ -167,27 +170,62 @@ export class DistributorDetector {
     provider: ethers.Provider,
     fromBlock: number,
     toBlock: number,
+    chunkSize: number = DEFAULT_BLOCK_CHUNK_SIZE,
   ): Promise<DistributorInfo[]> {
-    // Construct filter with OR logic for method signatures
-    const filter = {
-      address: ARBOWNER_PRECOMPILE_ADDRESS,
-      topics: [
-        OWNER_ACTS_EVENT_SIGNATURE,
-        [...ALL_DISTRIBUTOR_METHOD_SIGNATURES_PADDED],
-      ],
-      fromBlock,
-      toBlock,
-    };
+    const totalBlocks = toBlock - fromBlock + 1;
+    const needsChunking = totalBlocks > chunkSize;
 
-    // Query events with retry logic
-    const logs = await withRetry(() => provider.getLogs(filter), {
-      maxRetries: 3,
-      operationName: "scanBlockRange.getLogs",
-    });
+    if (needsChunking) {
```

Comment:
This should all be separated out into a separate utility. So that it can be reused in multiple places. We also shouldn't be logging any progress. If we want to add any logging like that, it'll be a separate ticket. The separate utility should obviously be more generalised.